src/components/nodeManager.js

```diff
@@ -1,46 +1,23 @@
 import { id as poolId } from './nodes/pool';
 import { id as laneId } from '@/components/nodes/poolLane';
 
-export function addIdToNodeAndSetUpDiagramReference(node, nodeIdGenerator) {
-  const [nodeId, diagramId] = nodeIdGenerator.generate();
-
-  if (!node.definition.id) {
-    node.definition.id = nodeId;
-  }
-
-  if (node.diagram) {
-    node.diagram.id = diagramId;
-    node.diagram.bpmnElement = node.definition;
-  }
-}
-
 export function addNodeToProcess(node, targetProcess) {
-  if (node.type === poolId || node.definition.$type === 'bpmn:MessageFlow') {
+  if (node.isType(poolId) || node.isBpmnType('bpmn:MessageFlow')) {
     return;
   }
 
-  if (node.type === laneId) {
+  if (node.isType(laneId)) {
     targetProcess
       .get('laneSets')[0]
       .get('lanes')
       .push(node.definition);
     return;
   }
 
-  if (node.definition.$type === 'bpmn:TextAnnotation' || node.definition.$type === 'bpmn:Association') {
-    targetProcess.get('artifacts').push(node.definition);
-    return;
+  let target = 'flowElements';
+  if (node.isBpmnType('bpmn:TextAnnotation') || node.isBpmnType('bpmn:Association')) {
+    target = 'artifacts';
   }
 
-  targetProcess.get('flowElements').push(node.definition);
-}
-
-export function getTargetProcess(node, processes, processNode) {
-  return node.pool
-    ? processes.find(({id}) => id === node.pool.component.node.definition.get('processRef').id)
-    : processNode.definition;
-}
-
-export function isBoundaryEvent(node) {
-  return node.definition.$type === 'bpmn:BoundaryEvent';
+  targetProcess.get(target).push(node.definition);
```

Comment:
I really wanted to make this function part of the node class too but kept running into a really weird problem. Simply moving the code into the node class without even calling it caused a very hard to diagnose crash. I spent a number of hours on this and sought help but still couldn't resolve it. I decided to leave this for now and to move on.